README.md

```diff
@@ -3,741 +3,470 @@
+
+**Fee-on-Transfer and Rebasing Token Handling:**
+-   **Fee-on-Transfer:** The Compact correctly handles fee-on-transfer tokens for both deposits and withdrawals. The amount of ERC6909 tokens minted or burned is based on the *actual balance change* in The Compact contract, not just the specified amount. This ensures ERC6909 tokens accurately represent the underlying assets.
+-   **Rebasing Tokens:** **Rebasing tokens (e.g., stETH) are NOT supported in The Compact V1.** Any yield or other balance changes occurring *after* deposit will not accrue to the depositor's ERC6909 tokens. For such assets, use their wrapped, non-rebasing counterparts (e.g., wstETH) to avoid loss of value.
+
+### Allocators
+Each resource lock is mediated by an **allocator**. Their primary responsibilities include:
+1.  **Preventing Double-Spending:** Ensuring sponsors don't commit the same tokens to multiple compacts or transfer away committed funds.
+2.  **Validating Transfers:** Attesting to standard ERC6909 transfers of resource lock tokens (via `IAllocator.attest`).
+3.  **Authorizing Claims:** Cosigning on claims against resource locks (via `IAllocator.authorizeClaim` or by providing off-chain `allocatorData`).
+4.  **Nonce Management:** Ensuring nonces are not reused for claims and (optionally) consuming nonces directly on The Compact using [`consume`](./src/interfaces/ITheCompact.sol#L550).
+
+Allocators must be registered with The Compact via [`__registerAllocator`](./src/interfaces/ITheCompact.sol#L561) before they can be assigned to locks. They must implement the [`IAllocator`](./src/interfaces/IAllocator.sol) interface.
+
+The trust assumptions around allocators are important:
+-   Claimants must trust that the allocator is sound and will not allow resource locks to become underfunded.
+-   Sponsors must trust that the allocator will not unduly censor valid requests against fully funded locks.
+
+### Arbiters
+Arbiters are responsible for verifying and submitting claims. When a sponsor creates a compact, they designate an arbiter who will:
+1.  Verify that the specified conditions of the compact have been met (these conditions can be implicitly understood or explicitly defined via witness data).
+2.  Process the claim by calling the appropriate function on The Compact (from [`ITheCompactClaims`](./src/interfaces/ITheCompactClaims.sol)).
+3.  Specify which accounts will receive the tokens and in what quantities as part of the claim payload.
+
+The trust assumptions around arbiters are also important:
+-   Sponsors must trust that the arbiter will not process claims where conditions weren't met.
+-   Claimants must trust that the arbiter will not fail to process claims where conditions *were* met.
+
+Often, the entity fulfilling an off-chain condition (like a cross-chain swap filler) might also act as or work closely with the arbiter.
```

Comment:
I'd phrase this as that the filler or solver might frequently *interface* directly with the arbiter